tests/helpers/test_condition.py

```diff
@@ -576,3 +578,18 @@ async def test_extract_devices():
             ],
         }
     ) == {"abcd", "qwer", "abcd_not", "qwer_not", "abcd_or", "qwer_or"}
+
+
+async def test_condition_template_error(hass, caplog):
+    """Test invalid template."""
+    caplog.set_level(ERROR)
+
+    test = await condition.async_from_config(
+        hass, {"condition": "template", "value_template": "{{ undefined.state }}"}
+    )
+
+    assert not test(hass)
+    assert len(caplog.records) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)